govuk-one-login/authentication-frontend
# AUT-1712: Remove code associated with supporting Zendesk

## What?

Removes code related to Zendesk support and constraints associated with previous Contact Centre on One Login contact forms.

## Why?

The new Contact Centre and SmartAgent are live in production, and we no longer need to support dual-running of Zendesk and the previous Contact Centre.

### High-level description of changes

At a high-level, changes include:

1. Deleting Zendesk specific services, interfaces and controller functions
2. Removing the code associated with forcing English locale within the support forms (this had been a requirement of the previous Contact Centre)
3. Removing the conditional routing and associated code that had allowed for SmartAgent and Zendesk to be supported in parallel across different environments
4. Updating environment variables
5. Renaming symbols to replace "zendesk" with a generic variant of "contact us"

### Detail of changes

The key changes and associated commits are listed below: 

* Add `SMARTAGENT_API_URL` to GitHub workflow env (commits: 9d8e8ab6 and a35cf95f)
* Set default for `getSmartAgentApiUrl` (commit: c44d26eb)
* Remove test for `setHtmlLangMiddleware` behaviour for contact us paths (commits: da12f6d2 and 9e45dbd)
* Remove Zendesk specific path from integration tests (commit: 322592a3)
* Delete `common-english-only` directory (commit: 9f617a52)
* Remove Zendesk specific env vars and getters (commit: 6c16d62b)
* Delete `getClientNameThatDirectsAllContactFormSubmissionsToSmartAgent` (commit: 7f6f55cb)
* Delete `setContactFormSubmissionUrlBasedOnClientName` and tests (commit: d4efcb63)
* Delete `CONTACT_US_TESTING_SMARTAGENT_IN_LIVE` from `PATH_NAMES` (commit: 36cc6e5b)
* Delete `CLIENT_NAME_THAT_DIRECTS_ALL_CONTACT_FORM_SUBMISSIONS_TO_SMARTAGENT` env var (commit: d1a33745)
* Remove `SUPPORT_SMART_AGENT` environment variable (commit: c347c271)
* Remove `SUPPORT_WELSH_LANGUAGE_IN_SUPPORT_FORMS` environment variable (commit: 50d605bb)
* Delete `ZendeskInterface` (commit: e79ce405)
* Delete Zendesk config (commit: c44c74b9)
* Delete `ZendeskService` (commit: 59273896)
* Rename multiple symbols to remove references to Zendesk (commit: b30f6af9)
* Delete `contactUsService` and `ContactUsServiceInterface` (commit: 2023ba35)
* Delete the redundant `contactUsQuestionsFormPostToZendesk` (commit: e78b3683)
* Remove conditional routing from contact us routes (commit: 33144c74)
* Remove functions associated with enforcing locale (commit: 8aa2c14e)
* Remove the `translateEnOnly` nunjucks filter (commits: 57fdb876 and 1f125e27)

### Manual testing of forms

While the SmartAgent integration should not be impacted by these changes, the code changes are so broad ranging that I've also manually tested the following sample of forms from my local machine to verify: content and error messages are rendered in the correct language; users are presented with the "Your message has been submitted" message upon submission and; submissions are being routed to the SmartAgent UAT environment (which is where my local development machine is configured to route submissions).

| Form path | Displays English content - `with(out)?` validation errors | Displays Welsh content - `with(out)?` validation errors | "Your message has been submitted" rendered | SmartAgent submission received (English) | SmartAgent submission received (Welsh)
|    :----:   |    :----:   |    :----:   |    :----:   |    :----:   |    :----:   |
| `/contact-us-questions?theme=account_creation&subtheme=no_security_code` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=signing_in&subtheme=account_not_found`| ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=something_else` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=id_check_app&subtheme=linking_problem` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=id_face_to_face&subtheme=face_to_face_details` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=proving_identity` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=email_subscriptions` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=suggestions_feedback` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=account_creation&subtheme=technical_error` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=id_face_to_face&subtheme=face_to_face_post_office_service`  | ✅ | ✅ | ✅ | ✅ | ✅ | 
 
I have also checked with QA to verify that this change will not impact our acceptance tests.
